### PR TITLE
Add footer with program version

### PR DESCRIPTION
**Pull request by levaitamas**
Implements #336. Looks super basic. Comments, edits, etc. are welcome. 

---

### Directory Listing:
image

---

### An Error Page:
image

---

### CODE REVIEW

File: src/renderer.rs [hunk 1]
```diff
@@ -138,12 +139,35 @@ pub fn page(
...
+                    }
                 }
             }
         }
     }
 }
 
+// Get version info
+fn get_version_info() -> String {
+    const PROG_NAME: Option<&'static str> = option_env!("CARGO_PKG_NAME");
```
Comment (svenstaro) [inline src/renderer.rs:153]: Small nit: You can use `crate_version!()` and `crate_name!()` from clap (which is re-exported from structopt which we already use). This should cut down on the code a little.

File: src/renderer.rs [hunk 2]
```diff
@@ -138,12 +139,35 @@ pub fn page(
...
+        "{}/{}",
+        PROG_NAME.unwrap_or("miniserve"),
+        VERSION.unwrap_or("unknown")
+    )
+}
+
+// Partial: version footer
+fn version_footer() -> Markup {
+    html! {
+        p style="text-align:center;padding-top:1em" {
```
Comment (svenstaro) [inline src/renderer.rs:165]: Let's make this a little less visible. I suggest making it smaller and giving it a somewhat succinct color (depending on theme).

The `at_color` in `style.scss` seems like a good candidate to build off of. I suggest make a new variable `footer_color` in every theme, use the respective values of `at_color` and just tone down the contrast a little.

**Review by svenstaro — CHANGES_REQUESTED** [review]
Very nice stuff. Just small nits. :)

**Comment by svenstaro** [comment]
Oh, and you can tell clippy to ignore the complexity for that function that's failing.

**Comment by levaitamas** [comment]
Thanks for your comments, @svenstaro. I managed to address them. Could you please take another look?

File: data/style.scss [hunk 3]
```diff
@@ -42,6 +42,12 @@ body {
     word-break: break-word;
 }
 
+.footer {
+    text-align: center;
+    padding-top: 1.5rem;
+    color: var(--footer_color);
```
Comment (svenstaro) [inline data/style.scss:48]: How about `font-size: 0.7em;` here as well?

Comment (levaitamas) [reply data/style.scss:49]: Sounds good. Done in 474fadd.

**Comment by svenstaro** [comment]
Great stuff, thanks! :)

**Pull request merged**